desci-server/src/controllers/users/associateWallet.ts

```diff
@@ -115,31 +127,31 @@ export const associateWallet = async (req: Request, res: Response, next: NextFun
       return;
     }
 
+    const user = (req as any).user;
     const message = new SiweMessage(req.body.message);
     const fields = await message.validate(req.body.signature);
-    if (fields.nonce !== (req as any).user.siweNonce) {
+    const siweNonce = await extractTokenFromCookie(req, 'siwe');
```

Comment:
need to do an additional check against DB to make sure nonce is not made up. in this setup anyone can make up a nonce and sign a msg that matches that nonce. need to make sure it also matches what's in the db @shadrach-tayo